client/src/Backoffice/components/Main/components/modals/EditDriverModal/EditDriverCarInfo/EditDriverCarInfo.tsx

```diff
@@ -0,0 +1,133 @@
+import { Checkbox, FormControl, FormControlLabel, FormHelperText, TextField } from '@mui/material';
+import { useFormContext } from 'react-hook-form';
+import { useEffect } from 'react';
+import { DRIVER_CAPABILITIES } from '../../../Volunteers/Volunteers.constants';
+import { DriverCarCapabilitiesEnum, Driver } from '../../../../../../../api-client/models/Driver';
+
+function EditDriverCarInfo({ driver }: { driver: Driver }) {
+  const {
+    register,
+    setValue,
+    formState: { errors }
+  } = useFormContext<Driver>();
+
+  useEffect(() => {
+    setValue('firstName', driver.firstName || '');
+    setValue('lastName', driver.lastName || '');
+    setValue('nationalId', driver.nationalId || '');
+  }, [driver, setValue]);
```

Comment:
When I tried it, modifying these values did not work. Whey did you need this useEffect?